### PR TITLE
Fix Ruby head matcher

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?
Fix Ruby Main in Ci

### What approach did you choose and why?

when using `assert_match` and a string is represented in the regexp pattern with two different opening/closing quotes(`')

The error in Ruby main before the fix
```
Expected /undefined method `render_component'/ to match # encoding: US-ASCII
#    valid: true
"undefined method 'render_component' for an instance of #<Class:0x0000000111930960>".
```

### Anything you want to highlight for special attention from reviewers?
we do not have Ruby main in Ci so it would be beneficial for reviewers to run the test suite against ruby head locally until it becomes a part of CI

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@boardfish I know you are working on more support for ruby 3.4 thats upcoming, can you take a look at this

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ nav_order: 5
 
     *Reegan Viljoen*
 
+* Fixed ruby head matcher issue
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:21]:
```suggestion
* Fixed ruby head matcher issue.
```

Comment (reeganviljoen) [reply docs/CHANGELOG.md]: thanks for the spot 👍

**Comment by joelhawksley** [comment]
Thanks @reeganviljoen!

**Comment by reeganviljoen** [comment]
> Thanks @reeganviljoen!

Its a pleasure 🎉

**Pull request merged**